lib/runner/extensions/request.command.js

```diff
@@ -114,17 +139,63 @@ module.exports = {
 
                 if (aborted) { return next(new Error('runtime: request aborted')); }
 
-                xhr = self.requester.request(item, function (err, response, request, cookies) {
-                    err = err || null;
-
-                    afterRequest(err, response, request, cookies);
+                Requester.create(context.trace || {
+                    type: 'http',
+                    source: 'collection',
+                    cursor: context.coords
+                }, self.options.requester, function (err, requester) {
+                    if (err) { return next(err); }  // this should never happen
+
+                    var sendId = RESPONSE_DOT + uuid();
+
+                    requester.on(sendId, self.triggers.io.bind(self.triggers));
+
+                    xhr = requester.request(sendId, item, function (err, res, req, cookies) {
+                        err = err || null;
+
+                        var nextPayload = {
+                                response: res,
+                                request: req,
+                                cookies: cookies
+                            },
+
+                            // called when we want to complete this request.
+                            complete = function () {
+                                // trigger the request event.
+                                // @note -  we give the _original_ item in this trigger, so someone can do reference
+                                //          checking. Not sure if we should do that or not, but that's how it is.
+                                //          Don't break it.
+                                afterRequest(err, res, req, cookies);
+
+                                // the error is passed twice to allow control between aborting the error vs just
+                                // bubbling it up
+                                return next((err && abortOnError) ? err : null, nextPayload, err);
+                            };
+
+                        if (err) {
+                            return complete();
```

Comment:
The error should be passed in for consistency, the signature for complete will have to be changed as well.